packages/ui/src/components/va-tabs/VaTab/VaTab.vue

```diff
@@ -39,111 +40,173 @@
 </template>
 
 <script lang="ts">
-import { inject } from 'vue'
-import { Options, mixins, prop, Vue, setup } from 'vue-class-component'
-
-import ColorMixin from '../../../services/color-config/ColorMixin'
-import useKeyboardOnlyFocus from '../../../composables/useKeyboardOnlyFocus'
-import { RouterLinkMixin } from '../../../mixins/RouterLinkMixin/RouterLinkMixin'
+import {
+  computed,
+  ComputedRef,
+  defineComponent,
+  inject,
+  onBeforeUnmount,
+  onMounted,
+  Ref,
+  ref,
+  SetupContext,
+  unref,
+} from 'vue'
 import VaIcon from '../../va-icon'
-
-import { TabsServiceKey, TabsService } from '../VaTabs.vue'
-
-type Context = {
-  tabsService: TabsService;
-  hasKeyboardFocus: boolean;
-  keyboardFocusListeners: Record<string, () => void>;
-}
-
-class TabProps {
-  color = prop<string>({ type: String, default: undefined })
-  icon = prop<string>({ type: String, default: null })
-  label = prop<string>({ type: String, default: null })
-  disabled = prop<boolean>({ type: Boolean })
-  name = prop<string | number>({ type: [String, Number] })
-  tag = prop<string>({ type: String, default: 'div' })
+import { useRouterLink, useRouterLinkProps } from '../../../composables/useRouterLink'
+import useKeyboardOnlyFocus from '../../../composables/useKeyboardOnlyFocus'
+import { useColor } from '../../../composables/useColor'
+import { TabsViewKey } from '../types'
+import { TabsView } from '../VaTabs.vue'
+
+export interface TabComponent {
+  id: string | number | null,
+  tabElement: Ref<HTMLElement | null>,
+  name?: string | number,
+  tabIndexComputed: ComputedRef<number>,
+  isActive: Ref<boolean>,
+  isActiveRouterLink: ComputedRef<boolean>,
+  rightSidePosition: Ref<number>,
+  leftSidePosition: Ref<number>,
+  onTabClick: () => void,
+  onTabKeydown: () => void,
+  onFocus: () => void,
+  updateSidePositions: () => void,
 }
 
-const TabPropsMixin = Vue.with(TabProps)
-
-@Options({
+export default defineComponent({
   name: 'VaTab',
   components: { VaIcon },
   emits: ['click', 'keydown-enter', 'focus'],
-})
-export default class VaTab extends mixins(
-  RouterLinkMixin,
-  TabPropsMixin,
-  ColorMixin,
-) {
-  id: string | number | null = null
-  isActive = false
-  rightSidePosition = 0
-  leftSidePosition = 0
-  hoverState = false
-
-  context: Context = setup(() => {
-    const tabsService = inject(TabsServiceKey) as TabsService
 
+  props: {
+    ...useRouterLinkProps,
+    color: {
+      type: String,
+      default: '',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    disabled: {
+      type: Boolean,
+    },
+    name: {
+      type: [String, Number],
+    },
+    tag: {
+      type: String,
+      default: 'div',
+    },
+  },
+
+  setup: (props, { emit }: SetupContext) => {
```

Comment:
Redundant return type

Reply:
It's not return type it is type of context - the second argument of setup function.

Reply:
Oh right. It's still redundant though, defineCompoment already handles it.

Reply:
Removed.